File: two-sums/two-sums.js
function twoSum(numbers, target) {
    const response = []
    const map = {}

    for (i = 0; i < numbers.length; i++) {
        let number = numbers[i]
        let difference = target - number

        if (map.hasOwnProperty(difference)) {
            response.push(i, map[difference])
            break
        }

        map[number] = i
    }

    return response
}

File: two-sums/two-sums.py
def twoSum(numbers, target):
    response: List[int] = []
    map = {}

    for index, number in enumerate(numbers):
        difference = target - number
        if difference in map:
            response.append(index)
            response.append(map[difference])
            continue

        map[number] = index

    return response